internal/services/catalog_service.go

```diff
@@ -32,6 +32,190 @@ type CatalogService struct {
 	kubeClient   *kubernetes.Cluster
 }
 
+// ServiceClass is a service class managed by Service catalog
+type ServiceClass struct {
+	Hash        string
+	Name        string
+	Broker      string
+	Description string
+	kubeClient  *kubernetes.Cluster
+}
+
+type ServiceClassList []ServiceClass
+
+// ServicePlan is a service plan managed by Service catalog
+type ServicePlan struct {
+	Name        string
+	Description string
+	Free        bool
+}
+
+type ServicePlanList []ServicePlan
+
+// ListPlans returns a ServicePlanList of all available catalog service plans, for the named class
+func (sc *ServiceClass) ListPlans() (ServicePlanList, error) {
+	servicePlanGVR := schema.GroupVersionResource{
+		Group:    "servicecatalog.k8s.io",
+		Version:  "v1beta1",
+		Resource: "clusterserviceplans",
+	}
+
+	dynamicClient, err := dynamic.NewForConfig(sc.kubeClient.RestConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	labelSelector := fmt.Sprintf("servicecatalog.k8s.io/spec.clusterServiceClassRef.name=%s", sc.Hash)
+
+	servicePlans, err := dynamicClient.Resource(servicePlanGVR).
+		List(context.Background(),
+			metav1.ListOptions{
+				LabelSelector: labelSelector,
+			})
+
+	if err != nil {
+		return nil, err
+	}
+
+	result := ServicePlanList{}
+
+	for _, servicePlan := range servicePlans.Items {
+		spec := servicePlan.Object["spec"].(map[string]interface{})
+
+		externalName := spec["externalName"].(string)
+		description := spec["description"].(string)
+		isAFreePlan := spec["free"].(bool)
+
+		result = append(result, ServicePlan{
+			Name:        externalName,
+			Description: description,
+			Free:        isAFreePlan,
+		})
+	}
+
+	return result, nil
+
+	// spec:
+	//   clusterServiceBrokerName: minibroker
+	//   clusterServiceClassRef:
+	//     name: mongodb
+	//   description: NoSQL document-oriented database that stores JSON-like documents with
+	//     dynamic schemas, simplifying the integration of data in content-driven applications.
+	//   externalID: mongodb-4-0-12
+	//   externalName: 4-0-12
+	//   free: true
```

Comment:
remove comments?

Reply:
I had left them in as example of the structure the code is digging through to get the information.

Reply:
Removed with commit 88d5750c76298888fa30a556e25ae969088be806